onnxruntime/python/tools/transformers/fusion_rotary_attention.py

```diff
@@ -917,7 +1040,10 @@ def fuse(self, normalize_node, input_name_to_nodes, output_name_to_node):
         self.nodes_to_remove.extend(qkv_nodes[1:])
 
         if v_nodes != v_nodes_4:
-            self.nodes_to_remove.extend(v_nodes[:-1])
+            if add_v:
+                self.nodes_to_remove.extend(v_nodes[:-2])
+            else:
+                self.nodes_to_remove.extend(v_nodes[:-1])
```

Comment:
Suggestion: `self.nodes_to_remove.extend(v_nodes[:-1] if add_v is None else v_nodes[:-2])`

It's better if the most common behavior comes first.